tests/agent/test_agent.py

```diff
@@ -279,6 +279,74 @@ def test_gx_agent_run_handles_subscriber_error_on_close(
     agent.run()
 
 
+def test_gx_agent_updates_cloud_on_job_status_error(
+    subscriber, create_session, get_context, client, gx_agent_config, event_handler
+):
+    correlation_id = "4ae63677-4dd5-4fb0-b511-870e7a286e77"
+    url = (
+        f"{gx_agent_config.gx_cloud_base_url}/api/v1/organizations/"
+        f"{gx_agent_config.gx_cloud_organization_id}/agent-jobs/{correlation_id}"
+    )
+    job_started_data = UpdateJobStatusRequest(data=JobStarted()).json()
+    job_completed = UpdateJobStatusRequest(
+        data=JobCompleted(success=True, created_resources=[], processed_by="agent")
+    )
+    job_completed_data = job_completed.json()
+
+    async def redeliver_message():
+        return None
+
+    event = RunOnboardingDataAssistantEvent(
+        datasource_name="test-ds", data_asset_name="test-da", organization_id=uuid.uuid4()
+    )
+
+    end_test = False
+
+    def signal_subtask_finished():
+        nonlocal end_test
+        end_test = True
+
+    event_context = EventContext(
+        event=event,
+        correlation_id=correlation_id,
+        processed_successfully=signal_subtask_finished,
+        processed_with_failures=signal_subtask_finished,
+        redeliver_message=redeliver_message,
+    )
+    event_handler.return_value.handle_event.return_value = ActionResult(
+        id=correlation_id, type=event.type, created_resources=[]
+    )
+
+    def consume(queue: str, on_message: Callable[[EventContext], None]):
+        """util to allow testing agent behavior without a subscriber.
+
+        Replicates behavior of Subscriber.consume by invoking the on_message
+        parameter with an event_context.
+        """
+        nonlocal event_context
+        on_message(event_context)
+
+        # we need the main thread to remain alive until event handler has finished
+        nonlocal end_test
+        while end_test is False:
+            sleep(0)  # defer control
```

Comment:
`non-blocking`
I would suggest refactoring `Susbcriber.consume` (and HTTP mocking) code into separate, well-named fixtures. The inclusion of it here complicates what is otherwise a simple test